cdap-client/src/main/java/io/cdap/cdap/client/ApplicationClient.java

```diff
@@ -304,7 +311,8 @@ public void upgradeApplication(ApplicationId app, Set<String> artifactScopes, bo
    */
   public void delete(ApplicationId app)
     throws ApplicationNotFoundException, IOException, AccessException {
-    String path = String.format("apps/%s/versions/%s", app.getApplication(), app.getVersion());
+    // Changing to non version specific app deletion - since version specific app deletion isn't supported after LCM.
```

Comment:
It would be cleaner to take an ApplicationReference instead of an ApplicationId, but I don't think it's all that important given how this client is being used, and seeing how none of the other methods are using version specific endpoints.

Would still be good to open a jira to clean this up though.

Reply:
Created https://cdap.atlassian.net/browse/CDAP-20404 : will address in separate PR after this is merged.